statuslistener/statuslistener.go

```diff
@@ -116,15 +117,30 @@ func (avs *AvailabilityStatusListener) processEvent(statusMessage types.StatusMe
 		return
 	}
 
+	previousStatus, err := dao.GetAvailabilityStatusFromStatusMessage(tenant.Id, statusMessage.ResourceID, statusMessage.ResourceType)
+	if err != nil {
+		l.Log.Errorf("unable to get status availability: %s", err)
+		return
+	}
+
 	resource.TenantID = tenant.Id
 	resource.AccountNumber = tenant.ExternalTenant
-	err = (*modelEventDao).FetchAndUpdateBy(*resource, updateAttributes)
-
+	resultRecord, err := (*modelEventDao).FetchAndUpdateBy(*resource, updateAttributes)
 	if err != nil {
-		l.Log.Errorf("Update error in status availability: %s", err)
+		l.Log.Errorf("unable to update availability status: %s", err)
 		return
 	}
 
+	if previousStatus != statusMessage.Status {
+		emailInfo := service.EmailNotificationInfoFromResource(resultRecord, previousStatus)
+		if emailInfo != nil {
+			err = service.NotificationProducer.EmitAvailabilityStatusNotification(accountNumber, emailInfo)
```

Comment:
Rather than calling a method on a package variable directly I'd rather have this just be a function that uses a default var at the package level. Much like we do for `service.RaiseEvent(...)`

Reply:
I did it but to accomplish it - I call this method on struct  in other method at  package level in order to use `NotificationProducer` with interface so I can use for mocking it in tests.